orchagent/portsorch.cpp

```diff
@@ -3166,6 +3171,46 @@ void PortsOrch::doTask(NotificationConsumer &consumer)
         }
 
         sai_deserialize_free_port_oper_status_ntf(count, portoperstatus);
+    } else if (op == "media_change") {
+        Port port;
+        for(auto i: values)
+        {
+            if(fvField(i) == "port")
+            {
+                alias = fvValue(i);
+                if (!getPort(alias, port))
+                {
+                    SWSS_LOG_ERROR("Failed to get port object for port alias %s", alias.c_str());
+                    return;
+                }
+                break;
+            }
+        }
+        for(auto i: values)
+        {
+            vector<uint32_t> serdes_val;
+            sai_attr_id_t attr_id = 0;
+
+            serdes_val.clear();
+            if(fvField(i) == "preemphasis")
+            {
+                attr_id = SAI_PORT_ATTR_SERDES_PREEMPHASIS;
+            }
+            else if(fvField(i) == "idriver")
+            {
+                attr_id = SAI_PORT_ATTR_SERDES_IDRIVER;
+            }
+            else if(fvField(i) == "ipredriver")
+            {
+                attr_id = SAI_PORT_ATTR_SERDES_IPREDRIVER;
+            } 
+            else if(fvField(i) == "port") 
```

Comment:
Hi Sudharsan,

I am reviewing the SAI piece of this as well at Broadcom and one thing I am wondering is how you handle interface mode.  Along with pre-emphasis settings, for DACs we will want to set either CR/XFI where for copper we would want to program SR/SFI.  The transceiver compliance field is all we would need but that would need to get passed down to SAI to set the interface mode properly.

Reply:
Hi @wschwartz Currently I don't think SONiC has a mechanism to program the interface mode. This pull request is just to handle pre-emphasis related settings alone. The interface mode can later be extended in this logic or can be incorporated in some other different logic. For now without setting interface mode, the pre-emphasis settings alone work good in SONiC and if there is a future requirement we can create a new PR for the related settings.

Reply:
yes, it can. check this attribute. SAI_PORT_ATTR_MEDIA_TYPE

@wschwatz, do we think we can fix this media type in another PR, or it has to be this PR?

Reply:
Hi @lguohan @dgsudharsan  I think this could be extended in the future to support interface mode in a future PR.  With TH3 coming we have a new set of interface modes to support so its probably good to take time and investigate those.    I think @ben-gale is working to find another reviewer as well.